src/features/MainScreen/NotesList/index.tsx

```diff
@@ -48,6 +51,23 @@ export const NotesList: FC<NotesListProps> = () => {
 
 	const items = virtualizer.getVirtualItems();
 
+	// Loads more notes when reaching the bottom of the notes list
+	const dispatch = useAppDispatch();
+	const workspaceData = useWorkspaceData();
+	useEffect(() => {
+		if (!notes.length) return;
+
+		const lastVisibleIndex = items[items.length - 1].index;
+		if (lastVisibleIndex >= notes.length - 10) {
```

Comment:
10 is too small threshold that would slow down UX. User must never feel the list are lazy.
This is our inner performance optimization. We should build UX to make user feel all notes are at hand.

So we should tune parameters a bit. First of all, let's extend a "page" size up to 300 notes.
Next thing is threshold must be large. Let's say 33% of page size.
Last thing is we should not to calc the limit manually. Calc it via formula like `Math.ceil(pageSize * lazyLoadingThreshold)` where `lazyLoadingThreshold` is a float number between 0 and 1.